src/test/java/fi/otavanopisto/kuntaapi/test/ResourceMocker.java

```diff
@@ -0,0 +1,194 @@
+package fi.otavanopisto.kuntaapi.test;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.removeStub;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static org.junit.Assert.fail;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.EnumMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.github.tomakehurst.wiremock.client.MappingBuilder;
+import com.github.tomakehurst.wiremock.matching.UrlPattern;
+
+public class ResourceMocker<I, R> {
+
+  private static final String APPLICATION_JSON = "application/json";
+  private static final String CONTENT_TYPE = "Content-Type";
+
+  private EnumMap<MockedResourceStatus, MappingBuilder> statusLists = new EnumMap<>(MockedResourceStatus.class);
+  private boolean started = false;
+  private Map<I, MockedResource<R>> resources = new LinkedHashMap<>();
+  private Map<I, List<ResourceMocker<?, ?>>> subMockers = new LinkedHashMap<>();
+  
+  public void start() {
+    started = true;
+    
+    for (MockedResource<R> resource : resources.values()) {
+      stubFor(resource.getCurrentMapping());
+    }
+    
+    for (Entry<MockedResourceStatus, MappingBuilder> statusListEntry : statusLists.entrySet()) {
+      MappingBuilder mapping = statusListEntry.getValue();  
+      MockedResourceStatus status = statusListEntry.getKey();
+      mapping.willReturn(aResponse()
+        .withHeader(CONTENT_TYPE, APPLICATION_JSON)
+        .withBody(toJSON(getResources(status))));
+      stubFor(mapping);
+    }
+    
+    for (I id : subMockers.keySet()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Iterate over the "entrySet" instead of the "keySet". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2864)